File: VB/MyObjects.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Namespace ComplexProperty
	Public Class Person
		Inherits XPObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Private _name As String
		Public Property Name() As String
			Get
				Return _name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _name, value)
			End Set
		End Property

		Private _address As Address
		<Aggregated> _
		Public Property Address() As Address
			Get
				Return _address
			End Get
			Set(ByVal value As Address)
				SetPropertyValue("Address", _address, value)
			End Set
		End Property
	End Class

	Public Class Address
		Inherits XPObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Private _addressLine1 As String
		Public Property AddressLine1() As String
			Get
				Return _addressLine1
			End Get
			Set(ByVal value As String)
				SetPropertyValue("AddressLine1", _addressLine1, value)
			End Set
		End Property

		Private _city As String
		Public Property City() As String
			Get
				Return _city
			End Get
			Set(ByVal value As String)
				SetPropertyValue("City", _city, value)
			End Set
		End Property

		Private _zip As String
		Public Property Zip() As String
			Get
				Return _zip
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Zip", _zip, value)
			End Set
		End Property
	End Class
End Namespace

File: CS/MyObjects.cs
using System;
using DevExpress.Xpo;

namespace ComplexProperty {
	public class Person : XPObject {
        public Person(Session session) : base(session) { }

        string _name;
        public string Name {
            get { return _name; }
            set { SetPropertyValue("Name", ref _name, value); }
        }

        Address _address;
        [Aggregated]
        public Address Address {
            get { return _address; }
            set { SetPropertyValue("Address", ref _address, value); }
        }
	}

	public class Address : XPObject {
        public Address(Session session) : base(session) { }

		string _addressLine1;
        public string AddressLine1 {
            get { return _addressLine1; }
            set { SetPropertyValue("AddressLine1", ref _addressLine1, value); }
        }

        string _city;
        public string City {
            get { return _city; }
            set { SetPropertyValue("City", ref _city, value); }
        }

        string _zip;
        public string Zip {
            get { return _zip; }
            set { SetPropertyValue("Zip", ref _zip, value); }
        }
	}
}
